code/src/savefile.c

```diff
@@ -686,3 +691,43 @@ void SaveFile_EnforceHealthLimit(void) {
         gSaveContext.health = 0;
     }
 }
+
+u8 SaveFile_SwordlessPatchesEnabled(void) {
+    return gSettingsContext.shuffleMasterSword && !(gExtSaveData.masterSwordFlags & 1);
+}
+
+u8 SaveFile_BecomeAdult(void) {
+    // Normal behaviour checks for 0xFF
+    if (gSaveContext.adultEquips.buttonItems[0] == 0xFF) {
+        if (SaveFile_SwordlessPatchesEnabled()) {
+            // Equip functionless item if not supposed to get master sword equipped
+            gSaveContext.adultEquips.buttonItems[0] = 0xFE;
+        } else {
+            // Normal behaviour if supposed to get master sword equipped
+            return 0;
+        }
+    }
+
+    // If first time going adult update adult equips (equip master sword if in inventory)
+    if (gSaveContext.adultEquips.equipment == 0) {
+        gSaveContext.adultEquips.equipment = 0x1120 | (gSaveContext.equipment & 0x2);
+        if (gSaveContext.equipment & 0x2) {
+            gSaveContext.adultEquips.buttonItems[0] = 0x3C;
+        }
+    }
```

Comment:
It will run the original code if the master sword has already been found on the file and the B value is 0xFF (SaveFile_SwordlessPatchesEnabled() becomes false and return 0 goes to original code for becoming adult with 0xFF). I've added the line that sets equipment to the first condition to simulate adult reset before obtaining the master sword